[nobodyishappy] iP

nobodyishappy opened the pull request:
# Unknown ❔ 

> “Death may be the greatest of all human blessings.” – Socrates ([source](https://www.joincake.com/blog/quotes-about-death/#h_7231242141831628700587173))

Unknown allows you to **track** your task to keep you on **track**.

This is a ~~good~~ **AMAZING** app:
- Cool GUI
- Intuitive
- Compact

Follow the steps to try:
1. Download it from [here](https://github.com/nobodyishappy/ip/releases/download/Level-10/duke.jar).
2. Double-click it.
3. Add your task.
4. 🎉 **BAM** 🎉 
5. Life is much simpler.

**Perfected** features:
- [x] Add tasks (ToDos, Deadlines, Events)
- [x] Delete tasks
- [x] Save and Load
- [x] Mark tasks
- [x] Find specific tasks

# 

If you are a Java programmer, you can use it to practice Java too. Here's the `Launcher` method:
```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

nobodyishappy committed:
Edited Duke.java

nobodyishappy committed:
Added PrintingString and completed level 1

nobodyishappy committed:
Completed level 2 tasks

nobodyishappy committed:
Added Task.java
Completed level 3 tasks

nobodyishappy committed:
Completed level 4 tasks

nobodyishappy committed:
Tried automated testing

nobodyishappy committed:
Complete level 5 tasks

nobodyishappy committed:
Completed level 6 tasks

nobodyishappy committed:
Added the function to save and load the task list

nobodyishappy committed:
Merge branch 'branch-Level-7'

nobodyishappy committed:
Add date processing for deadline

nobodyishappy committed:
Merge branch 'branch-level-8'

nobodyishappy committed:
Split Duke.java into seperate classes

nobodyishappy committed:
Merge branch 'A-MoreOOP'

nobodyishappy committed:
Split classes into packages

nobodyishappy committed:
Merge branch 'A-Packages'

nobodyishappy committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

nobodyishappy committed:
Added JUnit testing files

nobodyishappy committed:
Merge branch 'branch-A-JUnit'

nobodyishappy committed:
Added openable jar file

nobodyishappy committed:
Added JavaDoc

nobodyishappy committed:
Fixed Coding Standards

nobodyishappy committed:
Added find function

nobodyishappy committed:
Merge branch 'branch-A-JavaDoc'

nobodyishappy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke/Parser/Parser.java
#	src/main/java/Duke/TaskList/TaskList.java

nobodyishappy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Duke/Ui/Ui.java

nobodyishappy committed:
Fixed merge errors and improved coding standards

aureliony commented on src/main/java/Duke/Task/ToDos/ToDos.java:
```diff
@@ -0,0 +1,32 @@
+ * This class contains the functions for todos tasks.
+ * @author Tang Hao Liang
+ */
+public class ToDos extends Task {
```
Should this class name be in singular or plural form?

aureliony commented on build.gradle:
```diff
@@ -0,0 +1,42 @@
+    dependsOn("distZip", "distTar")
+}
+
+run{
```
Style: missing space

aureliony commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,96 @@
+        Ui.showWelcome();
+        boolean finished = false;
+
+        while(!finished) {
```
Style: missing space

aureliony commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,96 @@
+        boolean finished = false;
+
+        while(!finished) {
+            try{
```
Style: missing space

aureliony commented on src/main/java/Duke/DukeException/DukeException.java:
```diff
@@ -0,0 +1,23 @@
+ * @author Tang Hao Liang
+ */
+public class DukeException extends Exception {
+    protected String error;
```
Would it be better to name this as errorMsg?

aureliony commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,96 @@
+                    Ui.printList(contains);
+                } else {
+                    throw new DukeException("Please do enter a new proper command.\n");
+                }
```
Should this block of code be in Parser.java instead?

aureliony commented on src/main/java/Duke/Task/Task.java:
```diff
@@ -0,0 +1,72 @@
+ * This class contains the functions for tasks.
+ * @author Tang Hao Liang
+ */
+public class Task {
```
Should this be an abstract class?

aureliony reviewed (commented):
Should each class be in its own package? Or should we group related classes together in one package?

Also, should class names be in plural or singular form?

nobodyishappy committed:
Edit code with checkstyle

nobodyishappy committed:
Add GUI for application

nobodyishappy committed:
Merge branch 'branch-Level-10'

acekhoon commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,72 @@
+    protected boolean isDone;
+
+    /**
+     * Constructor to update description for task.
```
Very trivial comment, but I think java code convention asks to start the header comment using verbs

acekhoon commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+            task.markAsUndone();
+            return Ui.printUnmark(task.toString());
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a number for the task that is on the list.\n");
```
maybe here you can split the long string into two parts (long line split)

acekhoon commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+            taskList.remove(num);
+            return Ui.printDelete(task.toString(), taskList.size());
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a number for the task that is on the list.\n");
```
Same for here as well, can split the line into two parts

acekhoon commented on src/main/java/duke/task/events/Events.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public String toString() {
+        return "  [E]" + super.toString() + " (from: " + from + " to: " + to + ")";
```
maybe for better formatting, could consider using String.format() for readability since it gives more sense on how the overall string looks like

acekhoon commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,92 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
```
again very trivial comment, but might be better if you space between io, time and util imports

nobodyishappy committed:
Update gradle file

nobodyishappy committed:
Add assert for error checking

Three different asserts has been added:
- Checking whether load list is accurate
- Checking whether event output is accurate
- Checking whether delete works

nobodyishappy committed:
Merge pull request #1 from nobodyishappy/branch-A-Assertions

Add assert for error checking

nobodyishappy committed:
Improve code quality

Changes to code quality:
- Reduce level of nest by one for event method in parser
- Combine appendToFile and arrayToFile
- Extract loop variable to avoid complicated expression

Bug fix:
- Fix issue with loading of marked status

nobodyishappy committed:
Merge pull request #2 from nobodyishappy/branch-A-CodeQuality

Improve code quality

nobodyishappy committed:
Update to deadline and event

Deadline and event can now take in specific timing of the day for the task to be completed or start respectively.

nobodyishappy committed:
Add help command to showcase the builtin commands

nobodyishappy committed:
Merge branch 'branch-C-Help'

nobodyishappy committed:
Update to UI and README

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Update Ui.png

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Rename Ui.PNG to Ui.png

nobodyishappy committed:
Update README.md

nobodyishappy committed:
Update README.md
